### PR TITLE
Home page redesign

### PR DESCRIPTION
**Pull request by DutcherNL**
Added a new interactive home page

### CODE REVIEW

File: user_interaction/templates/user_interaction/home_users.html [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                </div>
+            </div>
+            <div class="col-md-4">
+                {% if not member %}
+                    <div class="alert alert-warning">
+                        This account has no membership connected yet. Please contact the board with your username so they
+                        can connect your membership to this account. Until that is done, functionality will be limited.
+                    </div>
+                {% else %}
+                    {% if member.associationgroup_set.exist %}
```
Comment (EricTRL) [inline user_interaction/templates/user_interaction/home_users.html:52]:
```suggestion
                    {% if member.associationgroup_set.exists %}
```

File: user_interaction/templates/user_interaction/home_users.html [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                                        </div>
+                                        <div>
+                                            {{ group.name }}
+                                        </div>
+                                    </div>
+                                    <ul class="list-group list-group-flush">
+                                        <li class="list-group-item">
+                                            <div class="row">
+                                                <div class="col">
+                                                    Control board
```
Comment (EricTRL) [inline user_interaction/templates/user_interaction/home_users.html:79]:
```suggestion
                                                    Control Panel
```

File: user_interaction/templates/user_interaction/home_users.html [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    <p>
+                        Bugs are annoying. If you encountered one, please mark them on github or notify the board and
+                        we will send in our troops!
+                    </p>
+                    <a class="btn btn-outline-dark" href="https://github.com/esrg-knights/Squire/issues">Report bug</a>
+                </div>
+            </div>
+        </div>
+    </div>
+{% endblock %}
```
Comment (EricTRL) [inline user_interaction/templates/user_interaction/home_users.html:118]: Privacy policy notice is missing

File: user_interaction/templates/user_interaction/home_users.html [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                                            </li>
+                                        {% endfor %}
+                                    </ul>
+                                </div>
+                            {% endfor %}
+                        </div>
+                    {% endif %}
+                {% endif %}
+
+                <div>
```
Comment (EricTRL) [inline user_interaction/templates/user_interaction/home_users.html:107]: I suggest to leave the COVID-activity-registration-notice on here somewhere (just like it is mentioned on the not-logged-in homepage)

File: user_interaction/templates/user_interaction/home_non_authenticated.html [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        Are you new to the Knights and want to join a boardgame or roleplaying evening? Please send a mail to
+        the board. Otherwise log-in.
+    </p>
+    <hr>
+    <h4>Login</h4>
+    <p>
+        Log in with your user account. Don't have one yet? Register and inform the board so they can connect your membership.
+    </p>
+    <div>
+        {% url "core:user_accounts/login" as login_url %}
```
Comment (EricTRL) [inline user_interaction/templates/user_interaction/home_non_authenticated.html:46]: There's no "forgot password" link or anything. I feel like a button to the login page is good enough; there's no need to have this functionality on multiple pages and this basically saves just one click

File: user_interaction/templates/user_interaction/home_users.html [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            <div class="col-md-8">
+                <div class="row">
+                    <div class="col-12">
+                        <h3>
+                            Activities in the next 7 days
+                        </h3>
+                    </div>
+                    {% for activity_moment in activities %}
+                        <div class="col-12 col-md-6 col-xl-4 p-1">
+                            {% include "activity_calendar/snippet_activity_moment.html" %}
```
Comment (EricTRL) [inline user_interaction/templates/user_interaction/home_users.html:40]: I feel like three of these in a single row is a little tight; the blocks get rather thin and high

File: user_interaction/templates/user_interaction/home_users.html [hunk 7]
```diff
@@ -0,0 +1,118 @@
+
```
Comment (EricTRL) [inline user_interaction/templates/user_interaction/home_users.html:1]: I like the overview, but right now the Activities page has pretty much the same functionality (with the only difference being the 7 days vs 14 days "preview")

File: user_interaction/views.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    template_name = "user_interaction/home_users.html"
+
+    def get_context_data(self, **kwargs):
+        # Get a list of all upcoming activities
+        start_date = timezone.now() - timedelta(hours=6)
+        end_date = start_date + timedelta(days=7)
+
+        activities = []
+        for activity in Activity.objects.filter(published_date__lte=timezone.now()):
+            for activity_moment in activity.get_all_activity_moments(start_date, end_date):
```
Comment (EricTRL) [inline user_interaction/views.py:38]: Method name changes in #153

**Review by EricTRL — CHANGES_REQUESTED** [review]

File: user_interaction/templates/user_interaction/home_users.html (hunk 3 above)
Comment (DutcherNL) [reply user_interaction/templates/user_interaction/home_users.html:123]: Notification is only visible when not logged in. Don't really see a use of placing it as a reminder constantly when you already accepted them. That said, this made me realise of a bigger problem that all those documents are currently hidden away too much. So I made #155. Also, did you know that on the registration page we do not mention the privacy document?

File: user_interaction/views.py (hunk 8 above)
Comment (DutcherNL) [reply user_interaction/views.py:39]: Ik heb een notitie gemaakt, maar als je het niet erg vind wil ik deze PR voor #153 invoeren dus zou het daar gefixed moeten worden. Heb een todo toegevoegd als heads up.

File: user_interaction/templates/user_interaction/home_users.html (hunk 3 above)
Comment (EricTRL) [reply user_interaction/templates/user_interaction/home_users.html:123]: > Also, did you know that on the registration page we do not mention the privacy document?
Good point! The mention should be there somewhere.

File: user_interaction/views.py (hunk 8 above)
Comment (EricTRL) [reply user_interaction/views.py:39]: Helemaal prima! Dit was ook een soort notitie voor mij :P

**Review by EricTRL — APPROVED** [review]

**Pull request merged**